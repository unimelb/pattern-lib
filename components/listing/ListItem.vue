<template>
  <div :class="classes">
    <slot />
  </div>
</template>

<script>
export default {
  props: {
    cols: {
      type: String,
      default: '',
    },
    noPadding: {
      type: String,
      default: '',
    },
  },
  computed: {
    classes() {
      return {
        cell: true,
        [`cell--no-padding-${this.noPadding}`]: [
          'all',
          'mobile',
          'tab',
          'desk',
        ].includes(this.noPadding),
        [`cell--desk-${this.cols}`]: [
          '1of1',
          '1of2',
          '1of3',
          '2of3',
          '1of4',
          '3of4',
          '1of6',
          '1of12',
        ].includes(this.cols),
      };
    },
  },
};
</script>
