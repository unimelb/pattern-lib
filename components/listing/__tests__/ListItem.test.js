import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import ListItem from '../ListItem.vue';

expect.extend(toHaveNoViolations);

describe('ListItem', () => {
  it('should match snapshot', () => {
    const result = shallow(ListItem).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(ListItem).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
