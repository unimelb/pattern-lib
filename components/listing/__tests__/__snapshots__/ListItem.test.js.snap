// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should match snapshot 1`] = `
<div
  class="cell"
/>
`;
